### PR TITLE
Власов М. С. 381806-1 ЛР №1

### PR DESCRIPTION
**Pull request by vla5924**
### Pull Request Readiness Checklist

See details at [code review]( https://github.com/UNN-ITMM-Software/devtools-course-practice/blob/main/lab-guide/lab-1-setup.md#%D1%80%D0%B5%D1%86%D0%B5%D0%BD%D0%B7%D0%B8%D1%8F-%D0%BA%D0%BE%D0%B4%D0%B0)

- [x] Make sure that the testing has passed successfully on [GitHub Actions](https://github.com/UNN-ITMM-Software/devtools-course-practice/actions)
- [ ] Get two approve from your fellow students
- [ ] Take a review from one of the administrators

### CODE REVIEW

File: modules/complex-number/test/test_vlasov_maksim_complex_number.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+TEST(Vlasov_Maksim_ComplexNumberTest, can_create_zero_by_default) {
+    double re = 0.0;
+    double im = 0.0;
+
+    ComplexNumber c;
+
+    EXPECT_EQ(re, c.getRe());
+    EXPECT_EQ(im, c.getIm());
+}
```
Comment (dkurt) [inline modules/complex-number/test/test_vlasov_maksim_complex_number.cpp:15]: Test is a duplication of https://github.com/UNN-ITMM-Software/devtools-course-practice/blob/71be99d6bbacf812170add82020dffd0d1f1e534/modules/complex-number/test/test_simple.cpp#L8-L12

File: modules/complex-number/test/test_vlasov_maksim_complex_number.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+TEST(Vlasov_Maksim_ComplexNumberTest, can_be_copied) {
+    double re = 1.0;
+    double im = 2.0;
+
+    ComplexNumber c(re, im);
+    ComplexNumber c_copy(c);
+
+    EXPECT_EQ(c.getRe(), c_copy.getRe());
+    EXPECT_EQ(c.getIm(), c_copy.getIm());
```
Comment (dkurt) [inline modules/complex-number/test/test_vlasov_maksim_complex_number.cpp:35]: I think it's more robust to compare `c_copy.getRe()` with `re` (same for imaginary part)

File: modules/complex-number/test/test_vlasov_maksim_complex_number.cpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+TEST(Vlasov_Maksim_ComplexNumberTest, can_be_copy_assigned) {
+    double re = 1.0;
+    double im = 2.0;
+
+    ComplexNumber c(re, im);
+    ComplexNumber c_copy = c;
+
+    EXPECT_EQ(c.getRe(), c_copy.getRe());
+    EXPECT_EQ(c.getIm(), c_copy.getIm());
```
Comment (dkurt) [inline modules/complex-number/test/test_vlasov_maksim_complex_number.cpp:46]: Same story as above

File: modules/complex-number/test/test_vlasov_maksim_complex_number.cpp (hunk 1 above)
Comment (vla5924) [reply modules/complex-number/test/test_vlasov_maksim_complex_number.cpp]: Yes, and the next one as well. Removed.

File: modules/complex-number/test/test_vlasov_maksim_complex_number.cpp (hunk 2 above)
Comment (vla5924) [reply modules/complex-number/test/test_vlasov_maksim_complex_number.cpp]: My concern was to compare these parts _directly_, but I agree about robustness. Fixed.

File: modules/complex-number/test/test_vlasov_maksim_complex_number.cpp (hunk 3 above)
Comment (vla5924) [reply modules/complex-number/test/test_vlasov_maksim_complex_number.cpp]: Also fixed

**Review by vlad-gogov — APPROVED** [review]

**Comment by dkurt** [comment]
You may always ask a person from https://docs.google.com/spreadsheets/d/11dxBeegDOpkStVuHZJgpatZV4HqxIkdf6H4gnRpclSM/edit#gid=0 to review by writing a comment with @ sign. In example,

@dkurt, please review

**Review by taktaev-artyom — APPROVED** [review]
Great job

**Review by vlad-gogov — APPROVED** [review]
Not bad

**Review by MaksimShagov — APPROVED** [review]

**Pull request merged**